File: align-android/app/src/main/java/ro/align/app/webrtc/SignalingUrl.kt
package ro.align.app.webrtc

import java.net.URLEncoder
import java.nio.charset.StandardCharsets

/** Echo din `lib/webrtc/signaling.ts` → `signalingWsConnectUrl`. */
fun buildSignalingUrl(baseUrl: String, token: String): String {
    val trimmed = baseUrl.trim().trimEnd('/')
    val withPath = if (trimmed.endsWith("/ws") || trimmed.endsWith("/ws/")) trimmed else "$trimmed/ws"
    val enc = URLEncoder.encode(token, StandardCharsets.UTF_8.name())
    return "$withPath?token=$enc"
}

File: align-ios/Align/Network/SignalingURL.swift
import Foundation

/// Echo `lib/webrtc/signaling.ts` → `signalingWsConnectUrl`.
func buildSignalingURL(base: String, token: String) -> URL? {
    var trimmed = base.trimmingCharacters(in: .whitespacesAndNewlines)
    while trimmed.hasSuffix("/") { trimmed.removeLast() }
    let withWs: String
    if trimmed.hasSuffix("/ws") {
        withWs = trimmed
    } else {
        withWs = "\(trimmed)/ws"
    }
    var comp = URLComponents(string: withWs)
    var q = comp?.queryItems ?? []
    q.append(URLQueryItem(name: "token", value: token))
    comp?.queryItems = q
    return comp?.url
}
